### PR TITLE
update to 2.0

### PR DESCRIPTION
**Pull request by kyldvs**
+ Updates build process to be similar to other fb projects (like facebook/fbjs)
+ Use fbjs for common modules (invariant, emptyFunction)
+ Removes modules that shouldn't be used (like mixinEventEmitter)
+ Syncs internal changes to the BaseEventEmitter class and supporting modules

```
// test plan
gulp build
npm test
```

### CODE REVIEW

**Comment by kyldvs** [comment]
Hmm, I can't figure out npm. When running this from another project it installs these changes but `lib` is not usually present in `node_modules/fbemitter` making this unusable. (although sometimes `lib` is there randomly..)

```
npm install --save kyldvs/emitter
```

**Comment by zpao** [comment]
When installing from github, you don't get build artifacts. Basically for this and number of our other projects you can never install directly. You can clone locally and link but that has its own pains.

I'll take a look at the rest tomorrow.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fbemitter",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (zpao) [inline package.json]: Let's bump the version separately.

File: scripts/jest/environment.js [hunk 2]
```diff
@@ -1 +1,2 @@
+
```
Comment (zpao) [inline scripts/jest/environment.js]: Can probably delete this :)

**Pull request merged**